jerry-core/ecma/base/ecma-helpers.h

```diff
@@ -52,6 +52,36 @@
  */
 #define ECMA_SET_POINTER(field, non_compressed_pointer) MEM_CP_SET_POINTER (field, non_compressed_pointer)
 
+/**
+ * Convert ecma-string's contents to a cesu-8 string and put it into a buffer.
+ */
+#define ECMA_STRING_TO_UTF8_STRING(ecma_str_ptr, /**< ecma string pointer */ \
+                                   utf8_ptr, /**< [out] output buffer pointer */ \
+                                   utf8_str_size) /**< [out] output buffer size */ \
+  lit_utf8_size_t utf8_str_size; \
+  bool utf8_ptr ## _is_ascii; \
+  const lit_utf8_byte_t *utf8_ptr = ecma_string_raw_chars (ecma_str_ptr, &utf8_str_size, &utf8_ptr ## _is_ascii); \
+  bool utf8_ptr ## must_be_freed = false; \
```

Comment:
Just an idea: could we pass the  utf8_ptr ## must_be_freed bool to ecma_string_raw_chars? This would make utf8_ptr ## _is_ascii unnecessary. Not sure there would be any gain. So this is just a question rather than a request.